src/components/TimeCardPage/CommentModal.tsx

```diff
@@ -1,70 +1,92 @@
+import React, { useState, useContext } from "react";
+import { CommentType } from "./types";
+import { CommentSchema } from "src/schemas/RowSchema";
+import { createNewComment } from "./CellTypes/CommentCell";
+import { UserContext } from "./UserContext";
 
-export default function CommentModal() {
+// setComments can be used for comments or reports
+interface DailyCommentModalProps {
+  setComments: Function;
+  comments: CommentSchema[];
+  type: CommentType;
+}
 
-    const formik = useFormik({
-        // TODO: add options for supervisor to choose from for the report
-        initialValues: {
-            type: "",
-            remarks: "",
-        },
-        onSubmit: (values) => {
-            // TODO: utilize a request to send comment to the backend (data is stored in 'values')
-            // Have to figure out what new reqeusts will need to be implemented in order to send 
-            // this to the backend
-            alert(JSON.stringify(values, null, 2));
-        }
-    });
+export function DailyCommentModal({
+  setComments,
+  comments,
+  type,
+}: DailyCommentModalProps) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [remark, setRemark] = useState();
+  const user = useContext(UserContext);
 
-    const { isOpen, onOpen, onClose } = useDisclosure()
+  // stores the remark that was made
+  const handleRemarkChange = (e) => {
+    setRemark(e.target.value);
+  };
 
-    return (
-        <>
-            <Button onClick={onOpen}>Comments/Reports</Button>
-            <Modal
-                isOpen={isOpen}
-                onClose={onClose}
-                isCentered
-                colorScheme={"red"}
-            >
-                <ModalContent>
-                    <VStack spacing={4} divider={<StackDivider />}>
-                        <ModalHeader>Leave a comment/report</ModalHeader>
-                        <form onSubmit={formik.handleSubmit}>
-                            <label htmlFor="remarks">Remarks</label>
-                            <Input
-                                id="remarks"
-                                name="remarks"
-                                type="text"
-                                onChange={formik.handleChange}
-                                value={formik.values.remarks}
-                            />
-                            <label htmlFor="Type">Type</label>
-                            <Input
-                                id="type"
-                                name="type"
-                                type="text"
-                                onChange={formik.handleChange}
-                                value={formik.values.type}
-                                placeholder="Enter Comment/Report"
-                            />
-                            <button type='submit'>Submit</button>
-                        </form>
+  const handleSubmit = () => {
+    setComments([...comments, createNewComment(user, type, remark)]);
```

Comment:
There is alot of new schema functionality added on main branch and this could begin to be extended to the backend if you wanted to fledge this out - also totally fine to split this out into a new ticket.